include/sway/criteria.h

```diff
@@ -7,10 +7,11 @@
 #include "tree/view.h"
 
 enum criteria_type {
-	CT_COMMAND          = 1 << 0,
-	CT_ASSIGN_OUTPUT    = 1 << 1,
-	CT_ASSIGN_WORKSPACE = 1 << 2,
-	CT_NO_FOCUS         = 1 << 3,
+	CT_COMMAND                 = 1 << 0,
+	CT_ASSIGN_OUTPUT           = 1 << 1,
+	CT_ASSIGN_WORKSPACE        = 1 << 2,
+	CT_ASSIGN_WORKSPACE_NUMBER = 1 << 3,
```

Comment:
Could we keep these two as one enum value and do the `isdigit` check in `select_workspace`? Perhaps we could eventually make `workspace_by_name` do the `isdigit` check and call `workspace_by_number` for us.

Reply:
Not necessarily. You can't just do the check straight off because `workspace 1` has different behaviour to `workspace number 1`.

Perhaps we could save the `number` bit and check whether the name begins with that, but then that would prevent workspaces from being named 'number 1', i.e. it would cause `workspace 'number 1'` to have the same behaviour as `workspace number 1`, which it currently does not.

Admittedly very esoteric, so this could be changed. Would require an args split during command parsing though.

Reply:
I'm happy to roll with your current implementation.